operations/cascade_utils.go

```diff
@@ -66,27 +66,45 @@ func recursiveGetAllLevelsCascadeOperations(ctx context.Context, object types.Ob
 	return operations, nil
 }
 
-func GetObjectChildren(ctx context.Context, object types.Object, storage storage.Repository) (cascade.CascadeChildren, error) {
-	children := make(cascade.CascadeChildren)
-	cascadeObject, isCascade := cascade.GetCascadeObject(ctx, object)
-	if isCascade {
-		for childType, childCriteria := range cascadeObject.GetChildrenCriterion() {
-			list, err := storage.List(ctx, childType, childCriteria...)
+func ListCascadeChildren(ctx context.Context, childrenCriterion cascade.ChildrenCriterion, storage storage.Repository) (cascade.CascadeChildren, error) {
+	cascadeChildren := make(cascade.CascadeChildren)
+	for childType, criterionDisjunction := range childrenCriterion {
```

Comment:
a. it is very hard to understand what this function is doing
b. the new logic replaces a single DB query with multiple ones, how would it affect perfromance?